### PR TITLE
[FlagGems Operator Development Competition] Add ctc_loss operator

### PR DESCRIPTION
**Pull request by lcthej**
### PR Category
  [Operator] [OP Test] [Benchmark]

  ### Type of Change
  [New Feature] [Performance Optimization]

  ### Description
  Add a Triton implementation of `ctc_loss` with forward and backward support.

  This PR includes:
  - `ctc_loss.IntList` and `ctc_loss.Tensor` registration
  - padded and concatenated target layouts
  - tensor/list input and target lengths
  - reductions: `none`, `mean`, `sum`
  - `zero_infinity` handling
  - non-default blank label coverage
  - benchmark shapes for forward and backward
  - unit tests for direct and registered `F.ctc_loss` paths
  - Triton 3.2+ autotuner warning fix through `do_bench`

  ### Issue
  N/A

  ### Progress
  - [x] Change is fully covered by a UT.
  - [x] Benchmark is added.
  - [ ] Change is properly reviewed.

  ### Performance
  Validation:
  - `python3 -m py_compile ...`: passed
  - `flake8` with project pre-commit args: passed
  - `isort --check-only --profile black`: passed
  - `git diff --check`: passed
  - `pytest -q tests/test_ctc_loss.py --tb=short -W error::DeprecationWarning`: `17 passed`

  Core benchmark:
  - `pytest -q -s -x benchmark/test_ctc_loss.py --level core --warmup 5 --iter 5 --dtypes float32 --dtypes float16 --record log --tb=short -W error::DeprecationWarning`
  - Result: `2 passed`

  Observed speedups vs PyTorch in the latest core run:
  - forward fp32: about `1.89x` to `10.18x`
  - forward fp16: about `3.01x` to `9.12x`
  - backward fp32: about `1.37x` to `5.61x`
  - backward fp16: about `1.84x` to `6.67x`

### CODE REVIEW

**Comment by lcthej** [comment]
Adding a coverage/performance checklist for the RaceDetail test-completeness items, limited to the CUDA/NVIDIA evidence available for this PR.

### Coverage checklist

| Dimension | Covered in this PR |
| --- | --- |
| API surface | `torch.nn.functional.ctc_loss` through `ctc_loss.IntList` and `ctc_loss.Tensor` registration, including forward and backward |
| Target layouts | Padded targets and concatenated targets |
| Length inputs | Tensor/list input lengths and target lengths |
| Reductions | `none`, `mean`, and `sum` |
| Parameters / edge cases | `zero_infinity`, non-default blank label, direct path and registered `F.ctc_loss` path |
| Dtypes | float32 and float16 benchmark coverage |
| Benchmark coverage | Adds `benchmark/test_ctc_loss.py` and `benchmark/core_shapes.yaml` entry for forward/backward shapes |
| Validation commands | `python3 -m py_compile src/flag_gems/ops/ctc_loss.py tests/test_ctc_loss.py benchmark/test_ctc_loss.py`; flake8 with project pre-commit args; `isort --check-only --profile black`; `git diff --check`; `pytest -q tests/test_ctc_loss.py --tb=short -W error::DeprecationWarning`; `pytest -q -s -x benchmark/test_ctc_loss.py --level core --warmup 5 --iter 5 --dtypes float32 --dtypes float16 --record log --tb=short -W error::DeprecationWarning` |
| Validation result | Unit tests: `17 passed`; core benchmark: `2 passed` |

### Performance evidence

Observed speedups vs PyTorch in the latest core run:

| group | observed speedup range |
| --- | ---: |
| forward fp32 | about `1.89x` to `10.18x` |
| forward fp16 | about `3.01x` to `9.12x` |
| backward fp32 | about `1.37x` to `5.61x` |
| backward fp16 | about `1.84x` to `6.67x` |

Benchmark command:

```bash
pytest -q -s -x benchmark/test_ctc_loss.py --level core --warmup 5 --iter 5 --dtypes float32 --dtypes float16 --record log --tb=short -W error::DeprecationWarning
```

This is grouped performance evidence from the current PR body rather than a pasted full per-row log; the benchmark file/core-shape coverage is included so reviewers can reproduce the row-level table with the command above.

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
